packages/web-components/fast-foundation/src/horizontal-scroll/horizontal-scroll.ts

```diff
@@ -0,0 +1,363 @@
+        let lastStop: number = 0;
+        const scrollItems: Array<any> = [].slice
+            .call(this.children)
+            /* filter out non-default slots */
+            .filter(
+                (el: Element): boolean =>
+                    !el.getAttribute("slot") || el.getAttribute("slot") === "default"
+            );
+
+        /* RTL items will come in reverse offsetLeft */
+        const isRtl: boolean =
+            scrollItems.length > 1 &&
+            scrollItems[0].offsetLeft > scrollItems[1].offsetLeft;
+        this.isRtl = isRtl;
+
+        const stops: Array<number> = scrollItems.map(
+            (
+                { offsetLeft: left, offsetWidth: width }: any,
+                idx: number,
+                ary: Array<any>
+            ): number => {
+                const firstLtr: boolean = !isRtl && idx === 0;
+                const lastRtl: boolean = isRtl && idx === ary.length - 1;
+                const right: number = (left + width) * (isRtl ? -1 : 1);
+
+                /* Getting the final stop after the last item or before the first RTL item */
+                if (!isRtl || idx === 0) {
+                    lastStop = right;
+                }
+
+                /* Remove extra margin on first item and last RTL item */
+                if (lastRtl || firstLtr) {
+                    left = 0;
+                } else if (isRtl) {
+                    left = right;
+                }
+
+                return left;
+            }
+        );
+
+        stops.push(lastStop);
+
+        /* Sort to zero */
+        stops.sort((a, b) => Math.abs(a) - Math.abs(b));
+
+        this.scrollStops = stops;
+        this.setFlippers();
+    }
+
+    /**
+     * Returns the current scroll position of the scrollContainer
+     * @internal
+     */
+    private getScrollPosition(): number {
+        return this.scrollContainer.scrollLeft;
+    }
+
+    /**
+     * Sets the controls view if enabled
+     * @internal
+     */
+    private setFlippers(): void {
+        const position: number = this.getScrollPosition();
+        if (this.previousFlipper) {
+            this.previousFlipper.classList.toggle("disabled", position === 0);
+        }
+        if (this.nextFlipper) {
+            const lastStop: number = this.scrollStops[this.scrollStops.length - 1];
+            this.nextFlipper.classList.toggle(
+                "disabled",
+                this.isRtl
+                    ? position - this.width <= lastStop
+                    : position + this.width >= lastStop
+            );
+        }
+    }
+
+    /**
+     * Scrolls items to the left
+     * @public
+     */
+    public scrollToPrevious(): void {
+        const position: number = this.getScrollPosition();
+        const stops: Array<number> = this.scrollStops;
+        if (this.isRtl) stops.sort((a: number, b: number): number => b - a);
+        const current: number = stops.findIndex(
+            (stop: number, idx: number, ary: Array<number>): boolean =>
```

Comment:
nit: I think verbosity is preferred for index and array for readability.

Reply:
Updated.